src/operator/cudnn_batch_norm-inl.h

```diff
@@ -156,22 +157,22 @@ class CuDNNBatchNormOp : public Operator {
         << "use global statistics is not yet supported in CuDNNBatchNorm";
 
     Stream<gpu> *s = ctx.get_stream<gpu>();
-    Tensor<gpu, 4> x = in_data[cudnnbatchnorm::kData].get_with_shape<gpu, 4, real_t>(shape_, s);
-    Tensor<gpu, 4> dx = in_grad[cudnnbatchnorm::kData].get_with_shape<gpu, 4, real_t>(shape_, s);
-    Tensor<gpu, 4> dy = out_grad[cudnnbatchnorm::kOut].get_with_shape<gpu, 4, real_t>(shape_, s);
-    Tensor<gpu, 1> gamma =
-      in_data[cudnnbatchnorm::kGamma].get_with_shape<gpu, 1, real_t>(Shape1(shape_[1]), s);
-    Tensor<gpu, 1> dbeta =
-      in_grad[cudnnbatchnorm::kBeta].get_with_shape<gpu, 1, real_t>(Shape1(shape_[1]), s);
-    Tensor<gpu, 1> dgamma =
-      in_grad[cudnnbatchnorm::kGamma].get_with_shape<gpu, 1, real_t>(Shape1(shape_[1]), s);
-    Tensor<gpu, 1> save_mean =
-      out_data[cudnnbatchnorm::kMean].get_with_shape<gpu, 1, real_t>(Shape1(shape_[1]), s);
-    Tensor<gpu, 1> save_inv_var =
-      out_data[cudnnbatchnorm::kInvVar].get_with_shape<gpu, 1, real_t>(Shape1(shape_[1]), s);
-    float a = 1.0f;
-    float b = 0.0f;
-    float b_add = 1.0f;
+    Tensor<gpu, 4, DType> x = in_data[cudnnbatchnorm::kData].get_with_shape<gpu, 4, DType>(shape_, s);
+    Tensor<gpu, 4, DType> dx = in_grad[cudnnbatchnorm::kData].get_with_shape<gpu, 4, DType>(shape_, s);
+    Tensor<gpu, 4, DType> dy = out_grad[cudnnbatchnorm::kOut].get_with_shape<gpu, 4, DType>(shape_, s);
+    Tensor<gpu, 1, DType> gamma =
+      in_data[cudnnbatchnorm::kGamma].get_with_shape<gpu, 1, DType>(Shape1(shape_[1]), s);
+    Tensor<gpu, 1, DType> dbeta =
+      in_grad[cudnnbatchnorm::kBeta].get_with_shape<gpu, 1, DType>(Shape1(shape_[1]), s);
+    Tensor<gpu, 1, DType> dgamma =
+      in_grad[cudnnbatchnorm::kGamma].get_with_shape<gpu, 1, DType>(Shape1(shape_[1]), s);
+    Tensor<gpu, 1, DType> save_mean =
+      out_data[cudnnbatchnorm::kMean].get_with_shape<gpu, 1, DType>(Shape1(shape_[1]), s);
+    Tensor<gpu, 1, DType> save_inv_var =
+      out_data[cudnnbatchnorm::kInvVar].get_with_shape<gpu, 1, DType>(Shape1(shape_[1]), s);
+    DType a = static_cast<DType>(1.0f);
```

Comment:
I think we don't need to use static_cast according to section 2.6 in cudnn doc. You can double check that.